### PR TITLE
Remove SyntheticColumnHandleBuilder

### PR DESCRIPTION
**Pull request by kokosing**
Remove SyntheticColumnHandleBuilder

This entity was not adding much value, it had only single and simple
implementation. However it introduced complexity:
 - changed many constructors APIs
 - protected field DefaultJdbcMetadataFactory
 - allowed to introduce an override of SyntheticColumnHandleBuilder
   which on it is own is is way more complex that single package
   private method

### CODE REVIEW

**Review by hashhar — APPROVED** [review]
LGTM

Some of it matches https://github.com/trinodb/trino/pull/18984, I'll rebase that after this is merged.

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestDefaultJdbcMetadata.java [hunk 1]
```diff
@@ -400,6 +397,34 @@ public void testMultiGroupKeyPredicatePushdown()
...
 
+    @Test
+    public void testColumnAliasTruncation()
+    {
+        assertThat(createSyntheticColumn(column("column_0"), 999).getColumnName())
+                .isEqualTo("column_0_999");
+        assertThat(createSyntheticColumn(column("column_with_over_twenty_characters"), 100).getColumnName())
+                .isEqualTo("column_with_over_twenty_ch_100");
+        assertThat(createSyntheticColumn(column("column_with_over_twenty_characters"), Integer.MAX_VALUE).getColumnName())
+                .isEqualTo("column_with_over_tw_2147483647");
```
Comment (dominikzalewski) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestDefaultJdbcMetadata.java:408]: minor: are we using soft assertions to understand which test cases actually fail? (as we replaced separate test with a single one)

**Review by dominikzalewski — APPROVED** [review]

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestDefaultJdbcMetadata.java (hunk 1 above)
Comment (kokosing) [reply plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestDefaultJdbcMetadata.java:408]: What is soft assertion?

Comment (dominikzalewski) [reply plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestDefaultJdbcMetadata.java:408]: https://joel-costigliola.github.io/assertj/core/api/org/assertj/core/api/SoftAssertions.html

They don't fail on the first failure in the test, but try to go through all assertions in the test and only then show a report of what failed.

There's a TestNG equivalent, but I'm not an expert on TestNG ;)

Comment (kokosing) [reply plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestDefaultJdbcMetadata.java:408]: Thank you for reference. I think regular assertions are fine. We don't use soft assertions anywhere so far, and I think there is no reason to surprise people here by doing something different than elsewhere.

I also removed using `DataProvider`. Copying code in tests could be sometimes better over DRY, because code is more readable. Here to be super tester, we could extract each assertion to separate test method, because they are independent. However that would be overkill too due boilerplate code (`DataProvider` potentially would better in that case?). Putting assertion in single method is kind of middle group. We do that in Trino where we put related assertion to a single method. 

TL;DR; it is all about the taste and habits
